Priority Queue/SimplePriorityQueue.cs

```diff
@@ -423,7 +423,36 @@ public bool TryDequeue(out TItem first)
         }
 
         /// <summary>
-        /// Attempts to remove an item from the queue.  The item does not need to be the head of the queue.  
+        /// Removes the head of the queue (node with minimum priority; ties are broken by order of insertion), 
+        /// and sets it to first along with its priority.
+        /// Useful for multi-threading, where the queue may become empty between calls to Contains(), GetPriority() and Dequeue()
+        /// Returns true if successful; false if queue was empty
+        /// O(log n)
+        /// </summary>
+        public bool TryDequeue(out TItem first, out TPriority firstPriority)
```

Comment:
I think this is a duplicate method. Create method, that dequeues simple node (private bool TryDequeue(out SimpleNode node)) and use it in both TryDequeue overrides